core/msg.c

```diff
@@ -106,54 +110,59 @@ static int _msg_send(msg_t *m, kernel_pid_t target_pid, bool block, unsigned sta
             DEBUG("msg_send() %s:%i: Target %" PRIkernel_pid
                   " has a msg_queue. Queueing message.\n", RIOT_FILE_RELATIVE,
                   __LINE__, target_pid);
+
+#ifdef MODULE_CORE_THREAD_FLAGS
+            int wakeup = thread_flags_wake(target);
+#else
+            int wakeup = 0;
+#endif
             irq_restore(state);
-            if (sched_active_thread->status == STATUS_REPLY_BLOCKED) {
+            if (wakeup || me->status == STATUS_REPLY_BLOCKED) {
```

Comment:
`wakeup` seems only to be used when `MODULE_CORE_THREAD_FLAGS` is compiled in.
You could put this `if` also into the `ifdef` and get rid of the superfluous check for `wakeup` (which is always 0 in this case) when `MODULE_CORE_THREAD_FLAGS` is not compiled in.